packages/pentaho/src/Canvas/ToolbarTabs/ToolbarTabs.tsx

```diff
@@ -0,0 +1,372 @@
+import {
+  AddAlt,
+  CloseXS,
+  MoreOptionsHorizontal,
+} from "@hitachivantara/uikit-react-icons";
+
+import { HvCanvasPanelTab } from "../PanelTab";
+import { HvCanvasPanelTabs, HvCanvasPanelTabsProps } from "../PanelTabs";
+import { staticClasses, TAB_WIDTH, useClasses } from "./ToolbarTabs.styles";
+
+export { staticClasses as canvasToolbarTabsClasses };
+
+export type HvCanvasToolbarTabsClasses = ExtractNames<typeof useClasses>;
+
+const DEFAULT_LABELS = {
+  create: "Create new",
+  undefined: "Undefined",
+  close: "Close",
+  dropdownMenu: "Dropdown menu",
+};
+
+interface ToolbarTabsTab {
+  id: string;
+  label: string;
+  icon?: React.ReactNode;
+}
+
+interface CloseButtonProps extends Omit<HvButtonProps, "onClick"> {
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>, tabId: string) => void;
+}
+
+export interface HvCanvasToolbarTabsProps
+  extends HvBaseProps<HTMLDivElement, "onChange"> {
+  /** When controlled, defines the tabs. */
+  tabs?: ToolbarTabsTab[];
+  /** When uncontrolled, defines the initial tabs. */
+  defaultTabs?: ToolbarTabsTab[];
+  /** Id of the selected tab if it needs to be controlled. */
+  tab?: string;
+  /** Defines the icon to be placed before the label when a new tab is created. If not defined, no icon is used. */
+  icon?: React.ReactNode;
+  /** Callback triggered when a tab changes/is clicked. */
+  onTabChange?: (
+    event: React.SyntheticEvent | null,
+    tabId: string | null,
+  ) => void;
+  /** Callback triggered when the tabs change: new tab added, tab removed, tab reorder, and label updated. */
+  onChange?: (
+    event: React.MouseEvent<HTMLButtonElement> | React.SyntheticEvent,
+    tabs: ToolbarTabsTab[],
+  ) => void;
+  /** Props to pass to the create new button. */
+  createButtonProps?: Partial<HvButtonProps>;
+  /** Props to pass to the close buttons. */
+  closeButtonProps?: Partial<CloseButtonProps>;
+  /** An object containing all the labels. */
+  labels?: Partial<typeof DEFAULT_LABELS>;
+  /** A Jss Object used to override or extend the styles applied. */
+  classes?: HvCanvasToolbarTabsClasses;
+}
+
+const DROPDOWN_MENU_WIDTH = 64;
+
+/**
+ * A toolbar tabs component to use in a canvas context.
+ */
+export const HvCanvasToolbarTabs = forwardRef<
+  HTMLDivElement,
+  HvCanvasToolbarTabsProps
+>((props, ref) => {
+  const {
+    children,
+    className,
+    tab: tabProp,
+    icon: iconProp,
+    createButtonProps: createButtonPropsProp,
+    closeButtonProps: closeButtonPropsProp,
+    tabs: tabsProp,
+    defaultTabs: defaultTabsProp = [],
+    labels: labelsProp,
+    classes: classesProp,
+    onTabChange: onTabChangeProp,
+    onChange: onChangeProp,
+    ...others
+  } = useDefaultProps("HvCanvasToolbarTabs", props);
+
+  const { onClick: onClickCreateButton, ...createButtonProps } =
+    createButtonPropsProp ?? {};
+  const {
+    onClick: onClickCloseButton,
+    style: styleCloseButton,
+    ...closeButtonProps
+  } = closeButtonPropsProp ?? {};
```

Comment:
For no reason in particular, my thought process was the following:
- I added `closeButtonProps` and `createButtonProps` because I thought that maybe some users would want to pass some props to these buttons: for example change the variant (I didn't think this deserved a prop of its own).
- When adding `closeButtonProps` and `createButtonProps`, I noticed that it might be possible for users to pass a `onClick` or `style` prop and I didn't want them to override the ones we have: this I destructured the props.

Reply:
But I'll remove `closeButtonProps` and `createButtonProps`. Does seem like they are absolutely needed at the moment.